### PR TITLE
CAYC - tests

### PR DESCRIPTION
**Pull request by duncanp-sonar**
No product code changes

### CODE REVIEW

**Comment by duncanp-sonar** [comment]
Low-value changes to get rid of [code smells in tests](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=4585&id=sonarlint-visualstudio&open=AYmHrNzNvmXtN3y4uMos) in the feature branch.

File: src/Education.UnitTests/Layout/Logical/RuleInfoTranslatorTests.cs [hunk 1]
```diff
@@ -84,7 +84,7 @@ public void GetRuleDescriptionSections_RootCauseSectionSetsCorrectTitle(RuleIssu
                         GetRuleInfo(new[] { new DescriptionSection("root_cause", null) }, issueType), null)
                     .ToList();
 
-            descriptionSections.Should().HaveCount(1);
```
Comment (duncanp-sonar) [inline src/Education.UnitTests/Layout/Logical/RuleInfoTranslatorTests.cs:87]: The Fluent test analyzer thinks `HaveCount` should be used except if there is only one, in which case `ContainSingle` should be used. Doesn't seem particularly consistent...

File: src/IssueViz.Security.UnitTests/Hotspots/LocalHotspotTests.cs [hunk 2]
```diff
@@ -50,6 +50,6 @@ public void Ctor_VisualizationIsNull_Throws()
                 null,
                 null));
 
-        test.Should().Throw<ArgumentNullException>().Which.Message.Should().Contain("visualization");
+        test.Should().Throw<ArgumentNullException>().WithMessage("*visualization*");
```
Comment (duncanp-sonar) [inline src/IssueViz.Security.UnitTests/Hotspots/LocalHotspotTests.cs:53]: Complains that `WithMessage` should be used, but that means we have to add wildcards to the string to get the same effect as `Contain`...

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]
Should we care about code smells in tests? Some of these are subjective as well. For example, I had the same issue about async methods raised by the IDE, but chose to ignore it as I think this Task.CompletedTask copy paste looks worse

**Pull request merged**